### ISSUE TITLE
Azure OpenAI 4.x Models

### ISSUE THREAD

**Issue by JoeMayo** [JoeMayo/generellem#169]
This task started off as an upgrade to GPT 3.5. However, after some discussion in our weekly open-source meeting, we really should implement a new `ILlm` for each model. This will give developers the ability to configure whatever model they want.

[Azure OpenAI Models](https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/models)

The code for the 4.x models are based on newer SDKs and won't be the same as the 3.5. An easy way to get an example is to create a model in Azure, via Azure OpenAI Studio, visit Chat, and click the View Code button. You'll see the differences between that code and the current `ILlm` implementation, `AzureOpenAILlm`. 

To summarize, the goal is to keep `AzureOpenAILlm` (3.5-turbo) and add support for all the 4.x versions.
